### PR TITLE
Add utils to main object

### PR DESCRIPTION
**Pull request by paulomarg**
### WHY are these changes introduced?

Many of the utils functions relied on the global config being available. With it gone, those utils would break, so we'll need some way to call them while including the config

### WHAT is this pull request doing?

For now, we're creating a "global" utils object in the exported library object, but we'll later refactor these individually so they're placed in a better place semantically.

### CODE REVIEW

File: src/utils/hmac-validator.ts [hunk 1]
```diff
@@ -20,13 +19,10 @@ export function stringifyQuery(query: AuthQuery): string {
...
 
-export function generateLocalHmac({
-  code,
-  timestamp,
-  state,
-  shop,
-  host,
-}: AuthQuery): string {
+export function generateLocalHmac(
+  apiSecretKey: string,
```
Comment (mkevinosullivan) [inline src/utils/hmac-validator.ts:23]: Putting a note here that this is changed in another branch, so these values can be obtained from config, when `generateLocalHmac` is attached using a `create__(config)` type function.

File: src/utils/hmac-validator.ts [hunk 2]
```diff
@@ -35,24 +31,19 @@ export function generateLocalHmac({
...
     .digest('hex');
 }
 
-/**
- * Uses the received query to validate the contained hmac value against the rest of the query content.
- *
- * @param query HTTP Request Query, containing the information to be validated.
- */
-export default function validateHmac(query: AuthQuery): boolean {
+export function validateHmac(apiSecretKey: string, query: AuthQuery): boolean {
```
Comment (mkevinosullivan) [inline src/utils/hmac-validator.ts:39]: Putting a note here that this is changed in another branch, so these values can be obtained from config, when `validateHmac` is attached using a `create__(config)` type function.

File: src/utils/index.ts [hunk 3]
```diff
@@ -1,34 +1,40 @@
...
+import {createGetEmbeddedAppUrl} from './get-embedded-app-url';
 
-export default ShopifyUtils;
+export function shopifyUtils(config: ConfigInterface) {
+  return {
+    decodeSessionToken,
+    // deleteCurrentSession: createDeleteCurrentSession(config),
+    // deleteOfflineSession: createDeleteOfflineSession(config),
+    // loadCurrentSession: createLoadCurrentSession(config),
+    // loadOfflineSession: createLoadOfflineSession(config),
```
Comment (mkevinosullivan) [inline src/utils/index.ts:29]: To be added later?

File: src/utils/with-session.ts [hunk 4]
```diff
@@ -1,66 +1,68 @@
 import * as ShopifyErrors from '../error';
-import {Session} from '../auth/session';
+import {SessionInterface} from '../auth/session/types';
 import {GraphqlClient} from '../clients/graphql';
 import {RestClient} from '../clients/rest';
```
Comment (mkevinosullivan) [inline src/utils/with-session.ts:4]: Will need to be refactored after rebase, given these types no longer exist

**Review by mkevinosullivan — APPROVED** [review]

File: src/utils/index.ts (hunk 3 above)
Comment (paulomarg) [reply src/utils/index.ts:29]: Yes, these will be fixed as part of oauth, there are a few interdependencies there so I felt it was better to tackle them all at once.

**Pull request merged**